src/VisualStudio/Xaml/Impl/Features/Diagnostics/XamlDiagnosticReport.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Features.Diagnostics
+{
+    internal class XamlDiagnosticReport
```

Comment:
1. not sure why this is internal, but the typ eabove is public.
2.consider making these non-mutable.  not sure hwat value there is in mutable types here.

Reply:
Oops, the XamlDiagnostic type should also be internal. My bad.

---
In reply to: [516287026](https://github.com/dotnet/roslyn/pull/49145#discussion_r516287026) [](ancestors = 516287026)

Reply:
The question here is why these have setters

---
In reply to: [516292574](https://github.com/dotnet/roslyn/pull/49145#discussion_r516292574) [](ancestors = 516292574,516287026)